### ISSUE TITLE
Not warning about missing types field

### ISSUE THREAD

**Issue by ghostdevv** [bluwy/publint#44]
Take the following package.json

```json
{
	"exports": {
		".": {
			"require": "./dist/index.cjs",
			"import": "./dist/index.js"
		}
	},
}
```

It didn't warn about a missing top level `types` field or a missing `types` field in the exports map, it did however warn about a missing `types` field in the exports map when adding a top level types field

**Comment by bluwy** [comment]
This is kinda intentional because some libraries don't publish types and relies on `@types/*` packages instead. Or if they don't publish JS at all, e.g. normalizecss.

It does warn when there's a top-level `"types"` field and none in `"exports"` because it signals the intent that you _do_ ship types, but your library won't work with `"moduleResolution": "bundler"`, so publint fills in the gap of this missing compatibility.

**Comment by ghostdevv** [comment]
Ah that makes sense :pray:
